### PR TITLE
Deploy apps when running `flynn-host update`

### PR DESCRIPTION
**Pull request by lmars**
This will deploy all system apps (except for `postgres`) in a specific order, then will deploy all other apps which use the slugrunner image.

### CODE REVIEW

File: host/cli/update.go [hunk 1]
```diff
@@ -26,47 +32,68 @@ Update Flynn components`)
...
 		return err
 	}
 	remote, err := tuf.HTTPRemoteStore(args.String["--repository"], nil)
 	if err != nil {
+		log.Error("error creating remote TUF client", "err", err)
 		return err
 	}
 	client := tuf.NewClient(local, remote)
+
+	log.Info("updating TUF client")
```
Comment (titanous) [inline host/cli/update.go]: s/client/data/

File: updater/updater.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	if err != nil {
+		log.Error("error getting apps", "err", err)
+		return err
+	}
+	for _, app := range apps {
+		if app.System() {
+			continue
+		}
+		log := log.New("name", app.Name)
+		log.Info("starting deploy of app")
```
Comment (titanous) [inline updater/updater.go]: `starting deploy of app to update slugrunner`

File: controller/client/client.go [hunk 3]
```diff
@@ -60,6 +62,21 @@ func NewClientWithHTTP(uri, key string, httpClient *http.Client) (*Client, error
 
 // NewClientWithPin acts like NewClient, but specifies a TLS pin.
 func NewClientWithPin(uri, key string, pin []byte) (*Client, error) {
+	return NewClientWithPinConfig(uri, key, &pinned.Config{Pin: pin})
+}
+
+// NewClientWithPinDomain acts like NewClient, but specifies a TLS pin and
+// custom domain (useful for when the route does not resolve, e.g. in tests).
+func NewClientWithPinDomain(uri, key string, pin []byte, domain string) (*Client, error) {
```
Comment (titanous) [inline controller/client/client.go]: `NewClientWithPinAndDomain`

Comment (titanous) [inline controller/client/client.go]: I wonder if we should just have a client config struct instead of the less common builder funcs?

**Comment by lmars** [comment]
@titanous comments addressed

**Comment by titanous** [comment]
LGTM pending CI

**Pull request merged**